test/lint.js

```diff
@@ -61,43 +100,77 @@ function checkSchema(dataFilename) {
     var valid = ajv.validate(schema, data);
 
     if (valid) {
-      console.log('  JSON Schema – OK');
+      return false;
     } else {
-      hasErrors = true;
-      console.log('  JSON Schema – ' + ajv.errors.length + ' error(s)')
-      // console.log(betterAjvErrors(schema, data, ajv.errors, { indent: 2 }));
-
-      // Output messages by one since better-ajv-errors wrongly joins messages
-      // (see https://github.com/atlassian/better-ajv-errors/pull/21)
-      // Other issues with better-ajv-errors:
-      // - it feels better for performance to output messages one by one rather than a list
-      // - it seems to be losing some errors when output a list
-      ajv.errors.forEach(function(error) {
-        var message = betterAjvErrors(schema, data, [error], { indent: 2 });
-
-        console.log('\n    ' + message.replace(/\n/g, '\n    '));
+      console.error(`\x1b[31m  JSON Schema – ${ajv.errors.length} error(s)\x1b[0m`)
+      ajv.errors.forEach(e => {
+        // Output messages by one since better-ajv-errors wrongly joins messages
+        // (see https://github.com/atlassian/better-ajv-errors/pull/21)
+        // Other issues with better-ajv-errors:
+        // - it feels better for performance to output messages one by one rather than a list
+        // - it seems to be losing some errors when output a list
+        console.error(betterAjvErrors(schema, data, [e], {indent: 2}));
       });
+      return true;
     }
   }
 }
 
-dictPaths.forEach(function(dir) {
-  var absDir = path.resolve(path.join(__dirname, '..', dir));
+dictPaths.forEach(dir => {
+  const absDir = path.resolve(path.join(__dirname, '..', dir));
 
-  fs.readdirSync(absDir).forEach(function(filename) {
+  fs.readdirSync(absDir).forEach(filename => {
     if (path.extname(filename) === '.json') {
-      var absFilename = path.join(absDir, filename);
+      let hasSyntaxErrors = false,
+        hasSchemaErrors = false,
+        hasStyleErrors = false;
 
-      console.log(dir + '/' + filename);
+      const absFilename = path.join(absDir, filename);
+      const relativeFilePath = path.relative(process.cwd(), absFilename);
 
-      checkStyle(absFilename)
-      checkSchema(absFilename);
+      const spinner = ora({
+        stream: process.stdout,
+        text: relativeFilePath
+      });
 
-      console.log();
+      const console_error = console.error;
+      console.error = (...args) => {
```

Comment:
That's a bad idea to monkey patching a console method. `testStyle` and `testSchema` could return an array of errors if any instead of calling `console.log()`. Later we can concat these arrays and if a resulting array is not empty then there is a file with errors.

Reply:
Take this up&nbsp;with the&nbsp;maintainers of&nbsp;https://github.com/mdn/browser-compat-data/blob/9d92e6d7f88cfa4400423a39183c321d7a68c410/test/lint.js#L52-L58